lib/Doctrine/ORM/Tools/SchemaTool.php

```diff
@@ -496,8 +518,8 @@ private function gatherColumn($class, array $mapping, Table $table)
      */
     private function gatherRelationsSql($class, $table, $schema, &$addedFks, &$blacklistedFks)
     {
-        foreach ($class->associationMappings as $mapping) {
-            if (isset($mapping['inherited'])) {
+        foreach ($class->associationMappings as $id => $mapping) {
+            if (isset($mapping['inherited']) && array_search($id, $class->identifier) === false) {
```

Comment:
Should be using strict mode - 3rd parameter `true`.

Reply:
Isn't `in_array()` more appropriated here?

Reply:
Good point, although both should do. :)